### PR TITLE
Feat/24/all student tasks

### PR DESCRIPTION
**Pull request by ArtemGermanenko**

### CODE REVIEW

File: server/routes/student/task.js [hunk 1]
```diff
@@ -27,4 +26,12 @@ route.post('/fileUpload', upload.fields([]), (req, res) => {
   res.sendStatus(200);
 });
 
+route.get('/my-tasks', (req, res) => {
```
Comment (whybazinga) [inline server/routes/student/task.js]: I suppose you should rename this request
Maybe smth like "tasks-list" ?

File: server/db-middleware/task-api.js [hunk 2]
```diff
@@ -26,6 +68,33 @@ apiModule.getAssignmentById = function (assId, assProj, taskProj, teacProj, stud
     .populate('studentId', studProj);
 };
 
+apiModule.getAllStudentTasks = function (studId) {
+  const result = {
+    resolved: [],
+    assignment: [],
+  };
+
+  let groupsId = [];
```
Comment (whybazinga) [inline server/db-middleware/task-api.js]: You don't need this (*)

File: server/db-middleware/task-api.js [hunk 3]
```diff
@@ -26,6 +68,33 @@ apiModule.getAssignmentById = function (assId, assProj, taskProj, teacProj, stud
...
+
+  return getAssignmentsByStudent(studId)
+    .then((assignments) => {
+      result.assignment = assignments;
+    })
+    .then(() => getGroupByStudId(studId))
+    .then((groupId) => {
```
Comment (whybazinga) [inline server/db-middleware/task-api.js]: Why do you need this .then()?
And why do you need groupsId in general?

File: server/db-middleware/task-api.js [hunk 4]
```diff
@@ -26,6 +68,33 @@ apiModule.getAssignmentById = function (assId, assProj, taskProj, teacProj, stud
...
+      groupsId = groupId;
+    })
+    .then(() => getAssignmentByGroup(groupsId))
```
Comment (whybazinga) [inline server/db-middleware/task-api.js]: Just pass your groupId here

File: server/db-middleware/task-api.js [hunk 5]
```diff
@@ -26,6 +67,28 @@ apiModule.getAssignmentById = function (assId, assProj, taskProj, teacProj, stud
     .populate('studentId', studProj);
 };
 
+apiModule.getAllStudentTasks = function (studId) {
+  const result = {
+    resolved: [],
```
Comment (AntonGrigoriev) [inline server/db-middleware/task-api.js]: There is no sense in these default empty arrays.

**Comment by ArtemGermanenko** [comment]
@AntonGrigoriev fixed

**Pull request merged**